apis/python/tests/test_collection.py

```diff
@@ -155,8 +157,10 @@ def test_delete_add(soma_object, tmp_path: pathlib.Path):
 
     with soma.open(tmp_uri, "w", soma_type=soma.Collection) as update:
         del update["porkchop sandwiches"]
-        pytest.xfail(reason="currently does not support removal/re-addition")
-        update["porkchop sandwiches"] = soma_object
+        # TEMPORARY: This should no longer raise once TileDB supports replacing
+        # an existing group member.
+        with pytest.raises(soma.SOMAError):
+            update["porkchop sandwiches"] = soma_object
```

Comment:
If this _method_ had the @pytest.xfail annotation, wouldn't that forgo the need to catch the exception? (And per `xfail`,  it would fail once it actually works, i.e. TileDB provides the necessary supports)

Reply:
The reason I want to avoid `xfail` on the method is because that will swallow *any* failure in the test (i.e., if something else breaks, it’s not visible). This instead verifies that only the part we want to not work doesn’t work. This should be temporary (replace support should come sooner than later).

This also highlights a weird mismatch between the two types of `xfail`: `@pytest.mark.xfail(...)` says “execute this, but if it fails that’s OK”, where `pytest.xfail()` in a test says “stop executing immediately mark the test xfailed”.